### PR TITLE
ci: migration gha pr prerelease

### PR DESCRIPTION
**Pull request by jportasa**
- [x] Add version print logic into codebase 
- [x] Add docker login for pull rate limitations 
- [x] Add snyk step as `needs` for prerelease once it will be ok
- [x] Migrate into Org secrets
- [x] Change upgrade codes for windows
- [x] Squash commits because Joan Porta did not accept CLA

### CODE REVIEW

**Comment by arvdias** [comment]
we have a small issue with MPL licenses in the dependencies. I've ignored them temporarily in Snyk, later we can see if we'll have an alternative or just ignore forever

File: .github/workflows/prerelease.yml [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+          password: ${{ secrets.OHAI_DOCKER_HUB_PASSWORD }}
+      - name: Pre release
+        run: make ci/prerelease
+      - name: Notify failure via Slack
+        if: ${{ failure() }}
+        uses: archive/github-actions-slack@master
+        with:
+          slack-bot-user-oauth-access-token: ${{ secrets.OHAI_SLACK_TOKEN }}
+          slack-channel: ${{ env.OHAI_SLACK_CHANNEL }}
+          slack-text: "Repo: ${{ env.REPO_FULL_NAME }}, prerelease for nix FAILURE!"
```
Comment (roobre) [inline .github/workflows/prerelease.yml:106]: `COREINT_` prefix on tokens, message format `❌ `${{ env.REPO_FULL_NAME }}`: prerelease pipeline failed.`

File: .github/workflows/prerelease.yml [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        shell: bash
+        run: |
+          build/windows/upload_msi.sh ${INTEGRATION} ${{ matrix.goarch }} ${TAG}
+      - name: Notify failure via Slack
+        if: ${{ failure() }}
+        uses: archive/github-actions-slack@master
+        with:
+          slack-bot-user-oauth-access-token: ${{ secrets.OHAI_SLACK_TOKEN }}
+          slack-channel: ${{ env.OHAI_SLACK_CHANNEL }}
+          slack-text: "Repo: ${{ env.REPO_FULL_NAME }}, prerelease for Windows FAILURE!"
```
Comment (roobre) [inline .github/workflows/prerelease.yml:149]: Prefix, format

File: build/windows/upload_msi.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+#
+#
+# Gets dist/zip_dirty created by Goreleaser and reorganize inside files
+#
+#
+INTEGRATION=$1
+ARCH=$2
+TAG=$3
+
+hub release edit -a "build/package/windows/nri-${ARCH}-installer/bin/Release/nri-${INTEGRATION}-${ARCH}.${TAG:1}.msi" -m ${TAG} ${TAG}
```
Comment (roobre) [inline build/windows/upload_msi.sh]: Replace with `gh` (https://github.com/newrelic/nri-mssql/commit/c126b3815244f309eecc60d0fea94eb9b40adf4d)

**Review by roobre — CHANGES_REQUESTED** [review]

File: .github/workflows/prerelease.yml (hunk 1 above)
Comment (arvdias) [reply .github/workflows/prerelease.yml]: I'll update the prefix after we get the proper secrets

File: .github/workflows/prerelease.yml [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+        with:
+          username: ${{ secrets.OHAI_DOCKER_HUB_ID }}
+          password: ${{ secrets.OHAI_DOCKER_HUB_PASSWORD }}
+      - name: Pre release
+        run: make ci/prerelease
+      - name: Notify failure via Slack
+        if: ${{ failure() }}
+        uses: archive/github-actions-slack@master
+        with:
+          slack-bot-user-oauth-access-token: ${{ secrets.OHAI_SLACK_TOKEN }}
```
Comment (paologallinaharbur) [inline .github/workflows/prerelease.yml:104]: we should remember to update slack naming

File: build/package/windows/nri-386-installer/Product.wxs [hunk 5]
```diff
@@ -1,11 +1,11 @@
 ﻿<?xml version="1.0" encoding="UTF-8"?>
 <Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
-    <?define Version = "{IntegrationVersion}" ?>
+    <?define Version = "$(var.IntegrationVersion)" ?>
     <?define ProductId = "*" ?>
-    <?define UpgradeCode = "{cbd764f3-feb5-4397-9902-a98c048ff9a3}" ?>
+    <?define UpgradeCode = "{62576008-3631-40a6-b34d-e5f02e99ebf6}" ?>
```
Comment (paologallinaharbur) [inline build/package/windows/nri-386-installer/Product.wxs:5]: the upgrade code should not change

File: build/package/windows/nri-amd64-installer/Product.wxs [hunk 6]
```diff
@@ -1,11 +1,11 @@
 ﻿<?xml version="1.0" encoding="UTF-8"?>
 <Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
-    <?define Version = "{IntegrationVersion}" ?>
+    <?define Version = "$(var.IntegrationVersion)" ?>
     <?define ProductId = "*" ?>
-    <?define UpgradeCode = "{cbd764f3-feb5-4397-9902-a98c048ff9a3}" ?>
+    <?define UpgradeCode = "{d61fc725-aabf-48d0-bf31-e22dbfd854af}" ?>
```
Comment (paologallinaharbur) [inline build/package/windows/nri-amd64-installer/Product.wxs:5]: same here

File: build/package/windows/nri-386-installer/Product.wxs (hunk 5 above)
Comment (paologallinaharbur) [reply build/package/windows/nri-386-installer/Product.wxs:5]:
```suggestion
    <?define UpgradeCode = "{cbd764f3-feb5-4397-9902-a98c048ff9a3}" ?>
```

File: build/package/windows/nri-amd64-installer/Product.wxs (hunk 6 above)
Comment (paologallinaharbur) [reply build/package/windows/nri-amd64-installer/Product.wxs:5]:
```suggestion
    <?define UpgradeCode = "{cbd764f3-feb5-4397-9902-a98c048ff9a3}" ?>
```

File: .github/workflows/prerelease.yml [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+          slack-channel: ${{ env.OHAI_SLACK_CHANNEL }}
```
Comment (paologallinaharbur) [inline .github/workflows/prerelease.yml:105]:
```suggestion
          slack-bot-user-oauth-access-token: ${{ secrets.COREINT_SLACK_TOKEN }}
          slack-channel: ${{ secrets.COREINT_SLACK_CHANNEL }}
```

File: .github/workflows/prerelease.yml [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+      - name: Upload MSI to GH
+        shell: bash
+        run: |
+          build/windows/upload_msi.sh ${INTEGRATION} ${{ matrix.goarch }} ${TAG}
+      - name: Notify failure via Slack
+        if: ${{ failure() }}
+        uses: archive/github-actions-slack@master
+        with:
+          slack-bot-user-oauth-access-token: ${{ secrets.OHAI_SLACK_TOKEN }}
+          slack-channel: ${{ env.OHAI_SLACK_CHANNEL }}
```
Comment (paologallinaharbur) [inline .github/workflows/prerelease.yml:148]:
```suggestion
          slack-bot-user-oauth-access-token: ${{ secrets.COREINT_SLACK_TOKEN }}
          slack-channel: ${{ secrets.COREINT_SLACK_CHANNEL }}
```

File: .github/workflows/prerelease.yml [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.OHAI_DOCKER_HUB_ID }}
+          password: ${{ secrets.OHAI_DOCKER_HUB_PASSWORD }}
+      - name: Scan code for vulnerabilities
+        env:
+          SNYK_TOKEN: ${{ secrets.COREINT_SNYK_TOKEN }}
```
Comment (roobre) [inline .github/workflows/prerelease.yml:45]: Same as elastic, move to job scope.

**Review by roobre — APPROVED** [review]
Same minor nitpick

**Review by paologallinaharbur — APPROVED** [review]

**Pull request merged**